example/lib/main.dart

```diff
@@ -116,6 +122,42 @@ class _MyAppState extends State<MyApp> {
                   customAttributes: {'key1': 'value1'},
                   customFlags: {'flag1': 'flagValue1'});
             }),
+            buildButton('Log Commerce - Product', () {
+              final Product product1 = Product('Orange', '123abc', 2.4, 1);
+              final Product product2 = Product('Apple', '456abc', 4.1, 2);
+              final TransactionAttributes transactionAttributes =
+                  TransactionAttributes('123456', 'affiliation', '12412342',
+                      1.34, 43.232, 242.2323);
+              mpInstance?.logCommerceEvent(
+                  productActionType: ProductActionType.Purchase,
+                  products: [product1.toJson(), product2.toJson()],
```

Comment:
I feel like `products` here should be a List<Product> rather than a List<String>, so then API is simpler:

```suggestion
                  products: [product1, product2],
```

it's then our job to take the `product` and `.toJson` it in the Dart layer.

The same applies to Promotion and Impression below.  Thoughts?